index.bs

```diff
@@ -3980,16 +3980,20 @@ When registering a new credential, represented by an {{AuthenticatorAttestationR
 
 1. Verify that the values of the [=client extension outputs=] in |clientExtensionResults| and the [=authenticator extension
     outputs=] in the <code>[=authdataextensions|extensions=]</code> in |authData| are as expected, considering the [=client
-    extension input=] values that were given in <code>|options|.{{PublicKeyCredentialCreationOptions/extensions}}</code>.
-    In particular, any [=extension identifier=] values in the |clientExtensionResults| and
-    the <code>[=authdataextensions|extensions=]</code> in |authData|
-    MUST also be present as [=extension identifier=] values in
-    <code>|options|.{{PublicKeyCredentialCreationOptions/extensions}}</code>,
-    i.e., no extensions are present that were not requested.
+    extension input=] values that were given in <code>|options|.{{PublicKeyCredentialCreationOptions/extensions}}</code>
+    and any specific policy of the [=[RP]=] regarding unsolicited extensions, i.e. those that were not specified as part of 
+    <code>|options|.{{PublicKeyCredentialCreationOptions/extensions}}</code>.
     In the general case, the meaning of "are as expected" is specific to the [=[RP]=] and which extensions are in use.
 
-    Note: Since all extensions are OPTIONAL for both the [=client=] and the [=authenticator=], the [=[RP]=] MUST be prepared to
-    handle cases where none or not all of the requested extensions were acted upon.
+    Note:  [=Client platforms=] MAY enact local policy that sets additional [=authenticator extensions=] or
+    [=client extensions=] and thus cause values to appear in the [=authenticator extension outputs=] or
+    [=client extension outputs=] that were not originally specified as part of 
+    <code>|options|.{{PublicKeyCredentialCreationOptions/extensions}}</code>. [=[RPS]=] should be aware of this and 
```

Comment:
```suggestion
    <code>|options|.{{PublicKeyCredentialCreationOptions/extensions}}</code>. [=[RPS]=] SHOULD be
```

Reply:
Or maybe this should even be a MUST? Since there's a normative MAY for clients/authenticators on the other end.

Reply:
The reason I originally didn't make this a "MUST", or even a "SHOULD", but left it as "should" is because what the RP does next is non-normative. For example an RP that decides to reject unsolicited extensions because that it its policy is just as conformant as one that decides to ignore them and accept the attestation or assertion. It'd be like saying:

RP's MUST be aware of this and can then do whatever they like about it...

What value is the "MUST" nomenclature in this situation?

I will have a go at a re-wording of this sentence, both for attestation and assertion validation, and see what you think after that....

Reply:
Actually when I look at the use of MUST in the note that immediately follows this one, it's the same thing. I'll go with your suggestion, but use MUST.

Reply:
Should be resolved in https://github.com/w3c/webauthn/pull/1289/commits/dc0378b9dd1aa142296ee25021e3b511e5b078c0

Reply:
Yeah, it's always a bit weird with the RP operations. But yeah, I think it's still worth making clear that the RP is _expected_ to be prepared for these situations - whether that means consulting some complex decision engine or just catching fire, so long as it's a deliberate decision.